src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs

```diff
@@ -371,25 +394,52 @@ where T: AsRef<[u8]>
 // -- if unwrap() will cause session panic.
 // -- https://github.com/chronotope/chrono/blob/v0.4.24/src/offset/mod.rs#L186
 // select to_date(to_timestamp('2021-03-28 01:00:00'));
-fn unwrap_local_time(tz: &Tz, d: &NaiveDate, times: &mut Vec<u32>) -> Result<DateTime<Tz>> {
-    match d.and_hms_opt(times[0], times[1], times[2]) {
-        Some(naive_datetime) => {
-            // Handling different cases for the timezone conversion
-            match tz.from_local_datetime(&naive_datetime) {
-                LocalResult::Single(t) => Ok(t),
-                LocalResult::None => Err(ErrorCode::BadBytes(format!(
-                    "Timezone conversion resulted in no valid time for tz: `{:?}`, date: {:?}, times: {:?}",
-                    tz, d, times
-                ))),
-                LocalResult::Ambiguous(t1, t2) => Err(ErrorCode::BadBytes(format!(
-                    "Ambiguous local time, ranging from {:?} to {:?}",
-                    t1, t2
-                ))),
+// Now add a setting force_timestamp_conversion to control this behavior. If true, try to add a hour.
+fn get_local_time(
+    tz: &Tz,
+    d: &NaiveDate,
+    times: &mut Vec<u32>,
+    force_timestamp_conversion: bool,
+) -> Result<DateTime<Tz>> {
+    d.and_hms_opt(times[0], times[1], times[2])
+        .map(|naive_datetime| unwrap_local_time(tz, force_timestamp_conversion, &naive_datetime))
+        .transpose()?
+        .ok_or_else(|| ErrorCode::BadBytes(format!("Invalid time provided in times: {:?}", times)))
+}
+
+fn unwrap_local_time(
```

Comment:
in src/query/functions/src/scalars/datetime.rs, has a same logic function: `convert_local_time`. But I do not known how to   

unify it in one place.

Reply:
databend-common-functions already deps on databend-common-io？why not pub this fn and call it one from datetime.rs